### PR TITLE
added support for 2.nightly 3.nightly and not specifying the repo wit…

### LINKED ISSUE #571: `using scala 3-nightly`
Scala-CLI allows defining the following using directive:

```scala
using scala "3"
```

which sets the Scala version to the latest stable version.

It would be a great improvement to be able to define a using directive like:

```scala
using scala 3-nightly
```

which would set the Scala version to the latest nightly.

### PR DESCRIPTION
**Pull request by ghost**
…h scala 2 nightly versions resolves #571

### CODE REVIEW

**Comment by ghost** [comment]
reviews from #650 are yet to be incorporated

**Comment by ghost** [comment]
I am incorporating the reviews of #650 in the #655  which edited the same functions defined and edited in #650. So if this gets merged, I can initiate a clean rebased PR for scala version error issue, which won’t include the nightly implementation, but would just attend to its corresponding issue with rewritten and concluded improvements from reviews of #650. @lwronski

File: modules/build/src/main/scala/scala/build/Artifacts.scala [hunk 1]
```diff
@@ -129,12 +129,23 @@ object Artifacts {
...
 
+    val scala2NightlyRepo = Seq(coursier.Repositories.scalaIntegration.root)
+
     val scalaNativeCliDependency =
       scalaNativeCliVersion.map(version =>
         Seq(dep"org.scala-native:scala-native-cli_2.12:$version")
       )
 
-    val allExtraRepositories = maybeSnapshotRepo ++ extraRepositories
+    val scala2NightlyRegex = raw"""2\.(\d+)\.(\d+)-bin-[a-f0-9]*""".r
```
Comment (lwronski) [inline modules/build/src/main/scala/scala/build/Artifacts.scala]: Move this regex to `ScalaParse.scala`.

**Review by lwronski — COMMENTED** [review]
LGTM, I added only one suggestion. 

You should also describe this feature in documentation.

File: modules/build/src/main/scala/scala/build/Artifacts.scala (hunk 1 above)
Comment (ghost) [reply modules/build/src/main/scala/scala/build/Artifacts.scala]: it is applied here:
https://github.com/VirtusLab/scala-cli/pull/655#issuecomment-1040123517

Comment (zmerr) [reply modules/build/src/main/scala/scala/build/Artifacts.scala:139]: modules/build/src/main/scala/scala/build/internal/ScalaParse.scala

File: website/docs/commands/compile.md [hunk 2]
```diff
@@ -57,6 +57,20 @@ scala-cli compile --scala 2 Hello.scala
 scala-cli compile --scala 3 Hello.scala
 ```
 
+## Scala Nightlies
+
+The nightly builds of Scala compiler are unstable ones which are published on a nightly basis.
+
+For using the latest scala 2 and scala 3 nightly builds, you should pass 2.nightly and 3.nightly, respectively.
```
Comment (lwronski) [inline website/docs/commands/compile.md]:
```suggestion
For using the latest scala 2 and scala 3 nightly builds, you should pass `2.nightly` and `3.nightly`, respectively. 
```

Comment (ghost) [reply website/docs/commands/compile.md]: applied

File: modules/build/src/main/scala/scala/build/options/BuildOptions.scala [hunk 3]
```diff
@@ -333,21 +341,79 @@ final case class BuildOptions(
...
+    either {
+      import coursier.Versions
+      import coursier.core.Latest
+      import coursier._
+      import scala.concurrent.ExecutionContext.{global => ec}
+
+      val moduleVersion: Option[String] = {
+        def scalaNightly2Module: Module = mod"org.scala-lang:scala-library"
+        val res = finalCache.logger.use {
+          Versions()
```
Comment (lwronski) [inline modules/build/src/main/scala/scala/build/options/BuildOptions.scala:393]: https://github.com/VirtusLab/scala-cli/pull/650#discussion_r805703410 - the same

File: modules/build/src/main/scala/scala/build/options/BuildOptions.scala [hunk 4]
```diff
@@ -333,21 +341,79 @@ final case class BuildOptions(
...
+        def scala3 = mod"org.scala-lang:scala3-library_3"
+        val res = finalCache.logger.use {
+          Versions()
+            .withModule(scala3)
+            .result()
+            .unsafeRun()(ec)
+        }
+        res.versions.latest(Latest.Release)
+      }
+      val scalaVersion       = moduleVersion.get
```
Comment (lwronski) [inline modules/build/src/main/scala/scala/build/options/BuildOptions.scala:375]: https://github.com/VirtusLab/scala-cli/pull/650#discussion_r805697823 - Therefore, I will ask you to move this changes to #650

Comment (ghost) [reply modules/build/src/main/scala/scala/build/options/BuildOptions.scala:375]: yes, I told you in detail in Slack that I have applied and listed all of them on #655. We resolved the issue on chat, and then you agreed to first merge this, and then #655. So why are we repeating the same resolved discussion again?

Comment (ghost) [reply modules/build/src/main/scala/scala/build/options/BuildOptions.scala:375]: https://github.com/VirtusLab/scala-cli/pull/655#issuecomment-1040123517

File: modules/build/src/main/scala/scala/build/options/BuildOptions.scala (hunk 3 above)
Comment (ghost) [reply modules/build/src/main/scala/scala/build/options/BuildOptions.scala:393]: As we discussed and concluded it in detail in Slack chat, they are applied all in here https://github.com/VirtusLab/scala-cli/pull/655#issuecomment-1040123517
So you told me that you decided to apply #652 and then #655. So what happened?

**Pull request closed without merging**